### PR TITLE
test: Testcase improvements for the CloudSqlInstanceTest

### PR DESCRIPTION
**Pull request by hessjcg**
Improves the stability of the CloudSqlInstanceTest and CloudSqlInstanceConcurrencyTest. This adds testcases
to cover refresh behavior for tokens near expiration. Also, it adds the PauseCondition so that the testcase can
orchestrate the refresh behavior across threads without resorting to `Thread.sleep()`.

### CODE REVIEW

File: core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java [hunk 1]
```diff
@@ -134,21 +135,150 @@ public void testCloudSqlInstanceForcesRefresh() throws InterruptedException {
...
+    // getSslData again, and assert the refresh operation completed.
+    SslData d2 = instance.getSslData();
+    assertThat(d2).isSameInstanceAs(data.getSslData());
+  }
+
+  @Test
+  public void testThatForceRefreshBalksWhenAForceRefreshIsInProgress() throws Exception {
+    InstanceData aboutToExpireData =
+        new InstanceData(
+            null, new SslData(null, null, null), Instant.now().plus(1, ChronoUnit.HOURS));
```
Comment (ttosta-google) [inline core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java:219]: it should be `ChronoUnit.SECONDS`

**Review by enocom — COMMENTED** [review]
In AlloyDB Java we use a deterministic scheduler to write tests like this. Could we do the same here?

File: core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java (hunk 1 above)
Comment (hessjcg) [reply core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java:219]: In this testcase, we want the initial data to have a long expiration, so that it does not immediately trigger a refresh. Then, we want to call forceRefresh() and see that exactly 1 refresh operation occurs. 

I have renamed the variable `initialData` instead of `aboutToExpireData` for clarity.

**Comment by hessjcg** [comment]
After some consideration, I think we should test concurrency using a ScheduledExecutor, while controlling the flow of execution using the PauseCondition. This approach actually exercises the concurrency behavior. Also, since our tests run across many OS and VM versions, it becamse easy to spot and fix subtle concurrency programming problems because a testcase would fail on only one or two OS+VM combo but not the others.

File: pom.xml [hunk 2]
```diff
@@ -153,6 +153,7 @@
           <compilerArgs>
             <arg>-XDcompilePolicy=simple</arg>
             <arg>-Xplugin:ErrorProne</arg>
+            <arg>-Xlint:-options</arg> <!-- ignore warning about deprecated java 8 src -->
```
Comment (enocom) [inline pom.xml:156]: nit: this would be nicer as separate PR just so people can discover it more easily.

**Review by enocom — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (hessjcg) [reply pom.xml:156]: I'll move it.

**Comment by enocom** [comment]
One other thought: we should port this to AlloyDB Java.

File: pom.xml (hunk 2 above)
Comment (hessjcg) [reply pom.xml:156]: See #1582
